framework/Models/ExternalModel.py

```diff
@@ -252,9 +252,14 @@ def evaluateSample(self, myInput, samplerType, kwargs):
     inRun = copy.copy(self._manipulateInput(Input[0][0]))
     result,instSelf = self._externalRun(inRun,Input[1],) #entry [1] is the external model object; it doesn't appear to be needed
     rlz = {}
-    rlz.update(inRun)
+    # FIXME: The results may contain the variables from input space, and
+    # the external model may change the values, we should raise
+    # an warning, and have a similar method as the code does
     rlz.update(result)
+    rlz.update(inRun)
     rlz.update(kwargs)
+    for k,v in rlz.items():
+      rlz[k] = np.atleast_1d(v)
```

Comment:
A better version of this is pending in the AdaptiveSobol, as well as the Optimizer merges.

Reply:
fixed.